### PR TITLE
SleepTimer - hackaton

### PR DESCRIPTION
**Pull request by osamamunirpk**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: resources/sleepTimer/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# AwsCommunity::any::sleepTimer
```
Comment (ericzbeard) [inline resources/sleepTimer/README.md:1]: Let's go with `AwsCommunity::Time::Sleep`. And let's create a Readme that describes the resource and provides a sample template.

File: resources/sleepTimer/awscommunity-any-sleeptimer.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+    "typeName": "AwsCommunity::any::sleepTimer",
+    "description": "An example resource schema demonstrating some basic constructs and validation rules.",
```
Comment (ericzbeard) [inline resources/sleepTimer/awscommunity-any-sleeptimer.json:3]: Update the description and source url

File: resources/sleepTimer/awscommunity-any-sleeptimer.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    "sourceUrl": "https://github.com/aws-cloudformation/aws-cloudformation-rpdk.git",
+    "definitions": {
+        "InitechDateFormat": {
```
Comment (ericzbeard) [inline resources/sleepTimer/awscommunity-any-sleeptimer.json:6]: Initech?

File: resources/sleepTimer/src/awscommunity_any_sleeptimer/handlers.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS,
+        resourceModel=model,
+    )
+    # TODO: put code here
+    print("Printed immediately.")
+    time.sleep(40)
```
Comment (ericzbeard) [inline resources/sleepTimer/src/awscommunity_any_sleeptimer/handlers.py:38]: This should be configurable via properties

File: resources/sleepTimer/src/awscommunity_any_sleeptimer/handlers.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    request: ResourceHandlerRequest,
+    callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+    progress: ProgressEvent = ProgressEvent(
+        status=OperationStatus.IN_PROGRESS,
+        resourceModel=model,
+    )
+    # TODO: put code here
+    return read_handler(session, request, callback_context)
```
Comment (ericzbeard) [inline resources/sleepTimer/src/awscommunity_any_sleeptimer/handlers.py:56]: We need to think about what the correct actions are for update, delete, etc. Do we always want the delay, or only for creates? Maybe this also should be configurable.

File: resources/sleepTimer/awscommunity-any-sleeptimer.json [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+            "$comment": "Use the `definitions` block to provide shared resource property schemas",
+            "type": "string",
+            "format": "date-time"
+        }
+    },
+    "properties": {
+        "Title": {
```
Comment (ericzbeard) [inline resources/sleepTimer/awscommunity-any-sleeptimer.json:13]: We need some configuration, like number of seconds to sleep, and possibly whether to sleep on all operations or just on create.

**Review by ericzbeard — CHANGES_REQUESTED** [review]

File: resources/sleepTimer/awscommunity-any-sleeptimer.json (hunk 6 above)
Comment (osamamunirpk) [reply resources/sleepTimer/awscommunity-any-sleeptimer.json:13]: I totally agree with above and great feedback.

**Pull request closed without merging**